contracts/script/deploy/mainnet/Mainnet_Deploy.s.sol

```diff
@@ -0,0 +1,547 @@
+            IStakeRegistry.StrategyParams[][] memory strategyAndWeightingMultipliers
+        ) = _parseStakeRegistryParams(config_data);
+        (
+            IRegistryCoordinator.OperatorSetParam[] memory operatorSetParams, 
+            address churner, 
+            address ejector
+        ) = _parseRegistryCoordinatorParams(config_data);
+
+        //upgrade the registry coordinator proxy to implementation
+        eigenDAProxyAdmin.upgradeAndCall(
+            TransparentUpgradeableProxy(payable(address(registryCoordinator))),
+            address(registryCoordinatorImplementation),
+            abi.encodeWithSelector(
+                RegistryCoordinator.initialize.selector,
+                eigenDAOwner,
+                churner,
+                ejector,
+                IPauserRegistry(pauser),
+                initalPausedStatus, 
+                operatorSetParams, 
+                minimumStakeForQuourm,
+                strategyAndWeightingMultipliers 
+            )
+        );
+        }
+
+        //deploy the eigenDA service manager implementation
+        eigenDAServiceManagerImplementation = new EigenDAServiceManager(
+            avsDirectory,
+            registryCoordinator,
+            stakeRegistry
+        );
+
+        //upgrade the eigenDA service manager proxy to implementation
+        eigenDAProxyAdmin.upgradeAndCall(
+            TransparentUpgradeableProxy(payable(address(eigenDAServiceManager))),
+            address(eigenDAServiceManagerImplementation),
+            abi.encodeWithSelector(
+                EigenDAServiceManager.initialize.selector,
+                IPauserRegistry(pauser),
+                initalPausedStatus,
+                deployer,
+                batchConfirmer
+            )
+        );
+
+        string memory metadataURI = stdJson.readString(config_data, ".uri");
+        eigenDAServiceManager.updateAVSMetadataURI(metadataURI);
+        eigenDAServiceManager.transferOwnership(eigenDAOwner);
+
+        //deploy the operator state retriever
+        operatorStateRetriever = new OperatorStateRetriever();
+
+        // transfer ownership of proxy admin to upgrader
+        eigenDAProxyAdmin.transferOwnership(eigenDAUpgrader);
+
+        vm.stopBroadcast();
+
+        // sanity checks
+        __verifyContractPointers(
+            apkRegistry,
+            eigenDAServiceManager,
+            registryCoordinator,
+            indexRegistry,
+            stakeRegistry
+        );
+
+        __verifyContractPointers(
+            apkRegistryImplementation,
+            eigenDAServiceManagerImplementation,
+            registryCoordinatorImplementation,
+            indexRegistryImplementation,
+            stakeRegistryImplementation
+        );
+
+        __verifyImplementations();
+        __verifyInitalizations(config_data);
+
+        //write output
+        _writeOutput(config_data);
+    }
+
+    function xtest() external {
+        // get info on all the already-deployed contracts
```

Comment:
can you make this call `run()` instead of copying the code? is the `require(deployer == tx.origin, ...)` check what's in the way right now?
I don't want to be too militant about code duplication but I find things to be much clearer + easier to review without copy-pasting 😄